src/inmanta/data/__init__.py

```diff
@@ -4481,23 +4486,23 @@ async def get_last_non_deploying_state_for_dependencies(
         if not resource_version_id.is_resource_version_id_obj():
             raise Exception("Argument resource_version_id is not a resource_version_id")
         query = """
-            SELECT r1.resource_version_id, r1.last_non_deploying_status
+            SELECT r1.resource_id, r1.model, r1.last_non_deploying_status
             FROM resource AS r1
             WHERE r1.environment=$1
                   AND r1.model=$2
                   AND (
                       SELECT (r2.attributes->'requires')::jsonb
                       FROM resource AS r2
-                      WHERE r2.environment=$1 AND r2.model=$2 AND r2.resource_version_id=$3
-                  ) ? r1.resource_version_id
+                      WHERE r2.environment=$1 AND r2.model=$2 AND r2.resource_id=$3
+                  ) ? r1.resource_id
```

Comment:
For this to work, don't we need to rewrite existing rows to store `resource_id` (rather than `resource_version_id` in `attributes.requires`? And if we do, then the `cleaned_requires` collection in `server/services/orchestrationservice.py` might be redundant.

Reply:
1. yes, we need to rewrite the DB
2. cleaned_requires is there precisely because we want to have resource_ids in the DB, while the api does RVID's